src/components/notifications/NotificationsMenu.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import { formatDistance, fromUnixTime } from "date-fns";
+import {
+    NOTIFICATIONS_KEY,
+    getNotifications,
+} from "./../../serviceFacades/notifications";
+import { useQuery } from "react-query";
+import { Skeleton } from "@material-ui/lab";
+import NotificationStyles from "./styles";
+import Divider from "@material-ui/core/Divider";
+import ids from "./ids";
+import { build } from "@cyverse-de/ui-lib";
+import Button from "@material-ui/core/Button";
+import {
+    ListItemText,
+    makeStyles,
+    Menu,
+    MenuItem,
+    Typography,
+} from "@material-ui/core";
+import { useTranslation } from "../../i18n";
+
+const useStyles = makeStyles(NotificationStyles);
+
+function getDisplayMessage(notification) {
+    if (notification) {
+        return notification.type === "data"
+            ? notification.subject
+            : notification.message.text;
+    }
+}
+
+function getTimeStamp(time) {
+    if (time) {
+        // slicing because time has extra zeroes in the unix string
+        const d = fromUnixTime(time.slice(0, -3));
+        return formatDistance(d, new Date());
+    }
+}
+
+function NotificationsMenu(props) {
+    const { notificationMssg } = props;
+    const [notifications, setNotifications] = useState([]);
+    const classes = useStyles();
+    const { t } = useTranslation(["common"]);
+
+    const handleClose = () => {
+        props.setAnchorEl();
+    };
+
+    useEffect(() => {
+        if (notificationMssg != null) {
+            setNotifications([notificationMssg, ...notifications].reverse());
+        }
+    }, [notifications, notificationMssg]);
+
+    const { isFetching } = useQuery({
+        queryKey: NOTIFICATIONS_KEY,
+        queryFn: getNotifications,
+        config: {
+            onSuccess: (results) => setNotifications(results?.messages),
+        },
+    });
+
+    return (
+        <Menu
+            anchorEl={props.anchorEl}
+            id={build(ids.BASE_DEBUG_ID, ids.NOTIFICATIONS_MENU)}
+            open={Boolean(props.anchorEl)}
+            onClose={handleClose}
+        >
+            <Typography
+                className={classes.header}
+                id={build(
+                    ids.BASE_DEBUG_ID,
+                    ids.NOTIFICATIONS_MENU,
+                    ids.NOTIFICATIONS_HEADER
+                )}
+                variant="h6"
+            >
+                {t("notifications")}
+            </Typography>
+            <Divider />
+
+            {isFetching && <Skeleton variant="rect" width={350} height={400} />}
+            {!isFetching &&
+                notifications.length > 0 &&
+                notifications.slice(0, 10).map((n, index) => (
```

Comment:
Ah, I missed this part in my review.  With what you have now, if the user gets 1,000 notifications (just as an extreme example), your component would be holding 1,000 notifications in state and growing with every new notification but only displaying 10.  

Instead of this, let's make this component more efficient and update state to only hold 10 notifications.

Reply:
The `slice` here can removed once you include my suggested commit.